lusSTR/cli/gui.py

```diff
@@ -197,14 +255,27 @@ def interactive_plots_allmarkers(sample_df, flagged_df):
     max_yvalue = (int(math.ceil(max_reads / n)) * n) + n
     increase_value = int(math.ceil((max_yvalue / 5) / n)) * n
     n = 0
-    for marker in sample_df["Locus"].unique():
+    all_loci = f_strs if st.session_state.kit == "forenseq" else p_strs
+    missing_loci = [x for x in all_loci if x not in sample_df["Locus"].unique()]
+    for marker in all_loci:
         col = cols[n]
         container = col.container(border=True)
         sample_locus = sample_df["SampleID"].unique() + "_" + marker
-        marker_df = sample_df[sample_df["Locus"] == marker].sort_values(by="CE_Allele")
+        for i, row in sample_df.iterrows():
+            if sample_df.loc[i, "Locus"] == "AMELOGENIN":
+                sample_df.loc[i, "CE_Allele"] = 0 if sample_df.loc[i, "CE_Allele"] == "X" else 1
```

Comment:
A little numpy action could clean this code up a bit.

```python
sample_df = np.where(
    sample_df["Locus"] == "AMELOGENIN",
    np.where(sample_df["CE_Allele"] == "X", 0, 1),
    sample_df["CE_Allele"],
)
```

--------

If you're not a fan of the functional approach and want to keep the row iteration, you may consider using the row object that you're currently ignoring.

```python
for i, row in sample_df.iterrows():
    if row["Locus"] == "AMELOGENIN":
        sample_df.loc[i, "CE_Allele"] = 0 of row.CE_Allele == "X" else 1
```

I can understand using `.loc` for all access and assignment operations, since it keeps the syntax consistent. But calling `.iterrows` without using the row objects is confusing on first read.